portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -62,18 +64,35 @@ public void init() {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-        // Create a query instance
-        Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
-
-        // Instantiate the datastore
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query queryAllComments = new Query("AllComments");
+        PreparedQuery resultsAllComments = datastore.prepare(queryAllComments);
+        Iterator<Entity> iterAllComments = resultsAllComments.asIterator();
+        Entity entityAllComments = iterAllComments.next();
+
+        Query queryComments = null;
+
+        // If the recent filter is selected, show most recent comments first 
+        if (entityAllComments.getProperty("filter").equals("recent")) {
+            // Create a query instance
+            queryComments = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+        } else if (entityAllComments.getProperty("filter").equals("oldest")) {
+            queryComments = new Query("Comment").addSort("time", SortDirection.ASCENDING);
+        } else if (entityAllComments.getProperty("filter").equals("top")) {
+            queryComments = new Query("Comment").addSort("popularity", SortDirection.DESCENDING);
+        } else if (entityAllComments.getProperty("filter").equals("bottom")) {
+            queryComments = new Query("Comment").addSort("popularity", SortDirection.ASCENDING);
+        } else if (entityAllComments.getProperty("filter").equals("alphabetical")) {
+            queryComments = new Query("Comment").addSort("name", SortDirection.ASCENDING);
+        } else {
+            Filter searchFilter = new FilterPredicate("name", FilterOperator.EQUAL, entityAllComments.getProperty("filter"));
+            queryComments = new Query("Comment").setFilter(searchFilter);
+        }
```

Comment:
Anand suggested to use switch-case here to make it a bit easier to read.

Reply:
Thanks! I pushed the commit that does this to my readability branch.